### PR TITLE
Refactored auth code to reuse what is there in track1 repo

### PR DESCRIPTION
**Pull request by vinagesh**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java [hunk 1]
```diff
@@ -44,18 +45,18 @@
      * @param connectionString The IoTHub connection string
      */
     @Builder(builderMethodName = "buildFromConnectionString", builderClassName = "FromConnectionStringBuilder")
-    public DigitalTwinAsyncClient(@NonNull String connectionString) {
-        ServiceConnectionString serviceConnectionString = ServiceConnectionStringParser.parseConnectionString(connectionString);
-        SasTokenProvider sasTokenProvider = serviceConnectionString.createSasTokenProvider();
-        String httpsEndpoint = serviceConnectionString.getHttpsEndpoint();
+    public DigitalTwinAsyncClient(@NonNull String connectionString) throws IOException {
+        IotHubConnectionString iotHubConnectionString = IotHubConnectionStringBuilder.createConnectionString(connectionString);
+        IotHubServiceSasToken iotHubServiceSasToken = new IotHubServiceSasToken(iotHubConnectionString);
```
Comment (abhipsaMisra) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:50]: do we get token caching and refresh?

Comment (vinagesh) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:50]: Good point. I will check.

Comment (vinagesh) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:50]: If you have context about this, let me know.

Comment (vinagesh) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:50]: Do you know if we had that in the copied code?

Comment (vinagesh) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:50]: Currently we have an expiry of 1 year for the SAS token so this implementation should be sufficient. We should revisit the logic of this in a separate task. I don't think maintaining code separately for DigitalTwinClient is the best option.

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java [hunk 2]
```diff
@@ -44,18 +45,18 @@
...
+        String httpsEndpoint = "https://" + iotHubConnectionString.getHostName();
```
Comment (drwill-ms) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:51]: is there not a way in Java to create a well formed URI? String concatenation makes me feel uneasy.

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java (hunk 1 above)
Comment (drwill-ms) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:50]: I'm curious about this too. Can we talk about it in parking lot or in a meeting?

Comment (vinagesh) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:50]: yes sure. I wanted to discuss with Abhipsa and the pnp team before bringing it to standup.

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java (hunk 2 above)
Comment (vinagesh) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java:51]: I don't see it done other ways. I will try to google some more. But would be great to know from others who code in java.

**Pull request closed without merging**